### PR TITLE
Replace KOKKOS_ENABLE_ISA_X86_64 with KOKKOS_USE_ISA_X86_64

### PR DESCRIPTION
**Pull request by masterleinad**
Both `CMake` and `Makefile.kokkos` only define `KOKKOS_USE_ISA_X86_64` (which is unused) but `KOKKOS_ENABLE_ISA_X86_64` is used instead. It appears that we wanted them to coincide.

### CODE REVIEW

**Comment by dalg24** [comment]
Did you consider doing defining `KOKKOS_ENABLE_ISA_X86_64` instead?

**Comment by masterleinad** [comment]
> Did you consider doing defining `KOKKOS_ENABLE_ISA_X86_64` instead?

Yes, I decided to go this way since the likelihood of someone else actually using the macro would be higher in the working form. Of course, from a uniformity point of view `KOKKOS_ENABLE_ISA_X86_64` might be better. What about defining both for a release or so and then only use the `ENABLE` variant?

File: cmake/KokkosCore_config.h.in [hunk 1]
```diff
@@ -26,6 +26,7 @@
 #ifndef __CUDA_ARCH__
 #cmakedefine KOKKOS_ENABLE_TM
 #cmakedefine KOKKOS_USE_ISA_X86_64
```
Comment (dalg24) [inline cmake/KokkosCore_config.h.in:28]:
```suggestion
#cmakedefine KOKKOS_USE_ISA_X86_64    // deprecated
```

File: cmake/KokkosCore_config.h.in [hunk 2]
```diff
@@ -25,7 +25,8 @@
 
 #ifndef __CUDA_ARCH__
 #cmakedefine KOKKOS_ENABLE_TM
-#cmakedefine KOKKOS_USE_ISA_X86_64
+#cmakedefine KOKKOS_USE_ISA_X86_64  // deprecated
+#cmakedefine KOKKOS_ENABLE_ISA_X86_64
 #cmakedefine KOKKOS_USE_ISA_KNC
 #cmakedefine KOKKOS_USE_ISA_POWERPCLE
 #cmakedefine KOKKOS_USE_ISA_POWERPCBE
```
Comment (dalg24) [inline cmake/KokkosCore_config.h.in:32]: These are also broken
https://github.com/kokkos/kokkos/blob/1410c2020a2a90412185a78d1d628826180cf798/core/src/impl/Kokkos_Core.cpp#L382-L394

**Comment by dalg24** [comment]
This needs more discussion.  I am not convinced we should keep the `KOKKOS_USE_` macro defines.  I don't see any use in Trilinos nor any mention in the wiki.
It is fair to ask if we should define the `KOKKOS_ENABLE_` macro at all.  It does not seem to be used anywhere else other than exporting the meta data with Kokkos Tools.

**Comment by masterleinad** [comment]
> It is fair to ask if we should define the `KOKKOS_ENABLE_` macro at all. It does not seem to be used anywhere else other than exporting the meta data with Kokkos Tools.

`KOKKOS_ENABLE_ISA_X86_64` is used in `core/src/impl/Kokkos_HostSpace.cpp` and `core/src/impl/Kokkos_Memory_Fence.hpp` (apart from the atomics) but the other `KOKKOS_ENABLE_ISA_*` not, right?

**Comment by masterleinad** [comment]
IMHO we should just fix `KOKKOS_ENABLE_ISA_X86` and discuss the removal of the other ones (and possibly the AVX preprocessor macros) somewhere else.

**Comment by MikolajZuzek** [comment]
Is `KOKKOS_USE_ISA_X86_64` allowed to be used in client applications (like Trilinos) ?

If it's internal to Kokkos and not used there, it can be removed, I guess. Looks like public [GitHub apps](https://github.com/search?q=KOKKOS_USE_ISA_X86_64+filename%3AKokkos_*&type=Code&ref=advsearch&l=&l=) do not use it except of in embedded Kokkos source code.

**Comment by ajpowelsnl** [comment]
Suggestion:  remove this macro

**Pull request closed without merging**